src/plugins/intel_gna/src/gna_graph_compiler.cpp

```diff
@@ -689,17 +782,38 @@ void GNAGraphCompiler::finalizeConvolution2DPrimitive(InferenceEngine::CNNLayerP
                                          convolution._kernel_y,
                                          convolution._kernel_x);
 
-    if (convolution._kernel_x > in_width || convolution._kernel_y > in_height) {
-        THROW_GNA_LAYER_EXCEPTION(layer) << "Kernel dimensions XY (" << convolution._kernel_x << ", "
-                                         << convolution._kernel_y << ")"
-                                         << " are bigger than input dimensions WH (" << in_width << "," << in_height
-                                         << ")";
+    // check if kernel need to be extended to stride size
+    const auto effective_kernel_h = std::max(convolution._kernel_y, convolution._stride_y);
+    const auto effective_kernel_w = std::max(convolution._kernel_x, convolution._stride_x);
+
+    auto effective_input_h = in_height;
+    auto effective_input_w = in_width;
+
+    // check if kernel input to be extended for new stride
+    const auto temp_input_h =
+        (out_height - 1) * convolution._stride_y - 2 * convolution._padding_y + effective_kernel_h;
+    const auto temp_input_w = (out_width - 1) * convolution._stride_x - 2 * convolution._padding_x + effective_kernel_w;
+
+    if (temp_input_h > effective_input_h) {
+        effective_input_h = temp_input_w;
+    }
+
+    if (temp_input_w > effective_input_w) {
+        effective_input_w = temp_input_w;
+    }
+
+    if (effective_kernel_w > effective_input_w || effective_kernel_h > effective_input_h) {
+        THROW_GNA_LAYER_EXCEPTION(layer) << "Kernel dimensions XY (" << effective_kernel_w << ", " << effective_kernel_h
+                                         << ")"
+                                         << " are bigger than input dimensions WH (" << effective_input_w << ","
+                                         << effective_input_h << ")";
     }
+
     const auto inputs = convolution.insData.front().lock();
     const auto outputs = convolution.outData.front();
 
     // have to pad input to let last kernel meets it's corresponding input
-    const auto num_inputs = in_width * in_height * in_channels;
+    const auto num_inputs = in_batch * effective_input_w * effective_input_h * in_channels;
     uint32_t num_input_padding = ALIGN(num_inputs, 8) - num_inputs;
```

Comment:
While doing changes here, please put this magic number into gna_limitations.

Reply:
There is already similar alignment in gna_limitations called
constexpr uint32_t inputByteAlignment = 64;
but it is equal to 64.

Reply:
As the name suggests it's related to byte alignment, not count alignment.

Reply:
will move 8 to the gna_limitations, bit agreed that it seems to be redundant for convolution2D